src/module-vsbridge-indexer-core/Index/IndexOperations.php

```diff
@@ -295,4 +321,100 @@ private function resolveClient($storeId): ClientInterface
     {
         return $this->clientResolver->getClient($storeId);
     }
+
+    /**
+     * @param $storeId
+     */
+    private function checkEsCondition($storeId)
+    {
+        $clusterHealth = $this->checkClustersHealth($storeId);
+        $this->checkMaxBulkQueueRequirement($clusterHealth, $storeId);
+    }
+
+    /**
+     * Check if clusters are in green status
+     *
+     * @param $storeId
+     *
+     * @return array|void
+     */
+    private function checkClustersHealth($storeId)
+    {
+        $clusterHealth = $this->resolveClient($storeId)->getClustersHealth();
+        if ($this->optimizationSettings->checkClusterHealth()) {
+            if ($clusterHealth[0]['status'] !== self::GREEN_HEALTH_STATUS) {
```

Comment:
run this ` $clusterHealth = $this->resolveClient($storeId)->getClustersHealth();`  only if it healt check is enable

so:
```
if ($this->optimizationSettings->checkClusterHealth()) {
        $clusterHealth = $this->resolveClient($storeId)->getClustersHealth();
```

Reply:
If you look to `function checkEsCondition($storeId)` you will know the reason of this structure. But maybe its bad that `checkEsCondition($storeId)` and `checkMaxBulkQueueRequirement($clusterHealth, $storeId)` are dependent on each other but I didn't want to send the request twice. I think that I can move `$clusterHealth = $this->resolveClient($storeId)->getClustersHealth();` to function `function checkEsCondition($storeId)` and will be better

Reply:
Yes, That what I have in mind :)

Reply:
Fixed